pinot-spi/src/main/java/org/apache/pinot/spi/executor/ExecutorServiceUtils.java

```diff
@@ -124,7 +124,7 @@ public static void close(ExecutorService executorService) {
   public static void close(ExecutorService executorService, long terminationMillis) {
     executorService.shutdown();
     try {
-      if (!executorService.awaitTermination(terminationMillis, TimeUnit.SECONDS)) {
+      if (!executorService.awaitTermination(terminationMillis, TimeUnit.MILLISECONDS)) {
```

Comment:
After shutdown it shouldn't be possible to submit any new tasks to ExecutorService and this will simply wait for existing tasks to finish.

Are you sure this is the root-cause for the issue you are seeing? Can you also confirm with a thread dump? (not a blocker for this PR though since the fix is required regardless)